L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc

```diff
@@ -745,8 +745,17 @@ void L1FPGATrackProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSe
     aTrack.setTrackWordBits();
     
     if (trackQuality_) {
-      trackQualityModel_->setL1TrackQuality(aTrack);
-      trackQualityDispModel_->setL1TrackQuality(aTrack);
+      double qualityScore = trackQualityModel_->getL1TrackQuality(aTrack);
+      aTrack.settrkMVA1(qualityScore);
+      if(extended_){
+	qualityScore = trackQualityDispModel_->getL1TrackQuality(aTrack);
+	if(trackQualityDispModel_->getQualityAlgorithm()==L1TrackQuality::QualityAlgorithm::GBDT){
+	  aTrack.settrkMVA2(qualityScore);
+	}
+	else{
+	  aTrack.settrkMVA1(qualityScore);
```

Comment:
This is not quite doing what we want. It seems like when extended==false, trkMVA1 is being overwritten with the qualityScore from the displaced model. We should remove the else statement, and only MVA2 is filled if extended_==true.

Also, why is there an additional constraint on the displaced model being a BDT? The displaced model should just be whatever is in the configuration and this seems like it will not let any changes to occur? I think it's better to make all changes to the configs than to the hardset code

Reply:
Currently only the BDT is defined for displaced tracks, right? I wanted to allow for the use of extended tracking with the prompt classifiers such as the NN. I've also found it useful to use both the prompt and displaced MVA score while using the extended tracking which is why I've set it up this way.

Reply:
That's right. I think we should always fill MVA1 with the prompt classifier no matter what (so that would be the removal of the else statement) and if someone wants to switch the prompt classifier to another one, say a NN, then that is done in the config. Then whenever extended tracking is run, we should fill MVA2 with the displaced one in addition to the prompt MVA1 one. And if people want to change the displaced one then that should happen in the config, no additional filtering should really be done here.

So here is what I'm imagining:
```
    if (trackQuality_) {
      aTrack.settrkMVA1(trackQualityModel_->getL1TrackQuality(aTrack));
      if(extended_)
	aTrack.settrkMVA2(trackQualityDispModel_->getL1TrackQuality(aTrack));
```

Reply:
Okay, I've made the fix [d95ffe9](https://github.com/cms-l1t-offline/cmssw/pull/1257/commits/d95ffe97501c7e003b3bc9de7e83685ecc20a6a6)